operators/src/util/raster_stream_to_geotiff.rs

```diff
@@ -344,22 +469,35 @@ impl<P: Pixel + GdalType> GdalDatasetWriter<P> {
         Ok(())
     }
 
-    fn finish(self) -> Result<()> {
+    fn finish(&mut self) -> Result<()> {
+        let input_dataset = self.reset_dataset().expect("dataset should be initialized");
+        let output_file_path = self
+            .result
+            .last()
+            .expect("finish should only be called after a temporal slice finished")
+            .params
+            .as_ref()
+            .expect("params should be set for a finished temporal slice")
+            .file_path
+            .clone();
```

Comment:
Can we do anything to not have to unwrap 3x in this statement?

This somehow sounds like a state machine, because you rely on state transitions :sweat_smile: .
Maybe this can be done by inserting 2 instead of one type? :thinking:

Reply:
- Clippy complained about too many lines. Since I had to do large scale refactoring anyway, I went the extra mile with write_no_data_value_grid and write_masked_data_grid. 358486707b7e9cd6b2d3017c9e19b4c69df97de1
- I removed the error::Error::EmptyInput again, since you said there is no way for raster queries to not have tiles.

Please note that 358486707b7e9cd6b2d3017c9e19b4c69df97de1 caused merge conflicts with the tile specification thing, so that needed to be untangled. 1190fa5b60e875a869de8b6b77ae2ac7b2b57493 a012c83feed0560f3f9eaf979b3e9f867bfc5932 5992963681eab30920b7d725d9646aed2f3a8453 95e78dd4f4931722d10ab8ebe521e50f2afd4823